pkg/config/msp_test.go

```diff
@@ -524,498 +527,753 @@ func TestAddRootCAToMSP(t *testing.T) {
-							},
-							"version": "0"
-						}
+							}
+						},
+						"version": "0"
 					},
-					"version": "0"
+					"Capabilities": {
+						"mod_policy": "Admins",
+						"value": {
+							"capabilities": {
+								"V1_3": {}
+							}
+						},
+						"version": "0"
+					}
 				},
-				"Orderer": {
-					"groups": {
-						"Org1": {
-							"groups": {},
-							"mod_policy": "Admins",
-							"policies": {
-								"Admins": {
-									"mod_policy": "Admins",
-									"policy": {
-										"type": 3,
-										"value": {
-											"rule": "MAJORITY",
-											"sub_policy": "Admins"
-										}
-									},
-									"version": "0"
-								},
-								"Endorsement": {
-									"mod_policy": "Admins",
-									"policy": {
-										"type": 3,
-										"value": {
-											"rule": "MAJORITY",
-											"sub_policy": "Endorsement"
-										}
-									},
-									"version": "0"
+				"version": "0"
+			}
+		},
+		"mod_policy": "",
+		"policies": {},
+		"values": {},
+		"version": "0"
+	},
+	"sequence": "0"
+}
+	`
+
+	expectedConfigProto := &cb.Config{}
+	err = protolator.DeepUnmarshalJSON(bytes.NewBufferString(expectedConfig), expectedConfigProto)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	gt.Expect(config).To(Equal(expectedConfigProto))
+}
+
+func TestAddRootCAToMSPFailure(t *testing.T) {
+	t.Parallel()
+	gt := NewGomegaWithT(t)
+
+	channelGroup, err := baseApplicationChannelGroup()
+	gt.Expect(err).ToNot(HaveOccurred())
+	config := &cb.Config{
+		ChannelGroup: channelGroup,
+	}
+
+	tests := []struct {
+		spec        string
+		cert        *x509.Certificate
+		expectedErr string
+	}{
+		{
+			spec: "invalid key usage",
+			cert: &x509.Certificate{
+				KeyUsage: x509.KeyUsageKeyAgreement,
+			},
+			expectedErr: "certificate KeyUsage must be x509.KeyUsageCertSign",
+		},
+		{
+			spec: "certificate is not a CA",
+			cert: &x509.Certificate{
+				IsCA:     false,
+				KeyUsage: x509.KeyUsageCertSign,
+			},
+			expectedErr: "certificate must be a CA certificate",
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.spec, func(t *testing.T) {
+			t.Parallel()
+			gt := NewGomegaWithT(t)
+			err = AddRootCAToMSP(config, tc.cert, "Org1")
+			gt.Expect(err).To(MatchError(tc.expectedErr))
+		})
+	}
+}
+
+func TestRevokeCertificateFromMSP(t *testing.T) {
+	t.Parallel()
+	gt := NewGomegaWithT(t)
+
+	channelGroup, err := baseApplicationChannelGroup()
+	gt.Expect(err).ToNot(HaveOccurred())
+	config := &cb.Config{
+		ChannelGroup: channelGroup,
+	}
+	org1MSP, err := GetMSPConfigurationForApplicationOrg(config, "Org1")
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(org1MSP.RevocationList).To(HaveLen(1))
+
+	caCert, caPrivKey := generateCACertAndPrivateKey("org1.example.com")
+	cert, _ := generateCertAndPrivateKeyFromCACert("Org1", caCert, caPrivKey)
+
+	err = RevokeCertificateFromMSP(config, "Org1", caCert, caPrivKey, cert)
+	gt.Expect(err).ToNot(HaveOccurred())
+
+	protolator.DeepMarshalJSON(os.Stdout, config)
```

Comment:
Why the marshal to Stdout? This is just going to create spam.

Reply:
Oops. I was just taking a look at the config and forgot to remove this line.